src/qiboml/operations/expectation.py

```diff
@@ -44,8 +46,9 @@ def expectation(
             ``qiboml.differentiation``.
     """
 
-    # read the frontend user choice
-    frontend = observable.backend
+    # read the qibo_backend user choice
+    qibo_backend = observable.backend
+    exec_backend = construct_backend(backend)
```

Comment:
I wouldn't construct the backend everytime, maybe it's better to directly ask for the backend object as input